libraries/aws/provisioning/src/aws_iot_provisioning_api.c

```diff
@@ -517,6 +516,195 @@ AwsIotProvisioningError_t AwsIotProvisioning_CreateKeysAndCertificate( IotMqttCo
 
 /*-----------------------------------------------------------*/
 
+AwsIotProvisioningError_t AwsIotProvisioning_CreateCertificateFromCsr( IotMqttConnection_t connection,
```

Comment:
Possibly refactor this function to reduce complexity as well as increase readability.  (Another hygiene type task, will eventually get addressed when you run complexity tool. Since this is new code, I'm pointing it out. )